src/Spryker/Zed/EventBehavior/Business/Model/EventResourceManager.php

```diff
@@ -0,0 +1,160 @@
+     */
+    protected $eventFacade;
+
+    /**
+     * @var \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[]
+     */
+    protected $eventResourcePlugins;
+
+    /**
+     * @var int
+     */
+    protected $chunkSize;
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Facade\EventBehaviorToEventInterface $eventFacade
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface[] $eventResourcePlugins
+     * @param int $chunkSize
+     */
+    public function __construct(
+        EventBehaviorToEventInterface $eventFacade,
+        array $eventResourcePlugins,
+        $chunkSize = 100
+    ) {
+        $this->eventFacade = $eventFacade;
+        $this->eventResourcePlugins = $eventResourcePlugins;
+        $this->chunkSize = $chunkSize;
+    }
+
+    /**
+     * @param array $resources
+     * @param array $ids
+     *
+     * @return void
+     */
+    public function triggerResourceEvents(array $resources, array $ids = [])
+    {
+        $this->mapPluginsByResourceName();
+        $plugins = $this->getEffectivePlugins($resources);
+
+        foreach ($plugins as $plugin) {
+            $this->triggerEvents($plugin, $ids);
+        }
+    }
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
+     * @param array $ids
+     *
+     * @return void
+     */
+    protected function triggerEvents(EventResourcePluginInterface $plugin, array $ids = [])
+    {
+        if ($ids) {
+            $this->trigger($plugin, $ids);
+
+            return;
+        }
+
+        if (!$plugin->queryData($ids)) {
+            $this->trigger($plugin, [static::ID_NULL]);
+
+            return;
+        }
+
+        $this->triggerEventsAll($plugin);
+    }
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
+     *
+     * @return void
+     */
+    protected function triggerEventsAll(EventResourcePluginInterface $plugin): void
+    {
+        $query = $plugin->queryData();
+        $count = $query->count();
+        $loops = $count / $this->chunkSize;
+        $offset = 0;
+
+        for ($i = 0; $i < $loops; $i++) {
+            $ids = $plugin->queryData()
+                ->offset($offset)
+                ->limit($this->chunkSize)
+                ->select([$plugin->getIdColumnName()])
+                ->find()
+                ->getData();
+
+            $this->trigger($plugin, $ids);
+            $offset += $this->chunkSize;
+        }
+    }
+
+    /**
+     * @param \Spryker\Zed\EventBehavior\Dependency\Plugin\EventResourcePluginInterface $plugin
+     * @param array $ids
+     *
+     * @return void
+     */
+    protected function trigger(EventResourcePluginInterface $plugin, array $ids): void
+    {
+        //TODO replace this with triggerBulk as soon this method is available in Event module
```

Comment:
At least remove this todo for now and create a subticket in DataImporterOptimization ticket, we will fix this there

Reply:
Created sub-task: https://spryker.atlassian.net/browse/PS-2866